src/plugins/intel_cpu/src/nodes/scaled_attn_softmax.cpp

```diff
@@ -0,0 +1,599 @@
+    auto v_scale = _mm512_set1_ps(scale);
+    auto v_a = v_max;
+    auto v_b = v_max;
+    size_t i = 0;
+    // process vector body
+    while (i + 16 <= size) {
+        v_a = _mm512_loadu_ps(a + i);
+        v_b = _mm512_loadu_ps(b + i);
+        v_a = _mm512_fmadd_ps(v_a, v_scale, v_b);
+        v_max = _mm512_max_ps(v_max, v_a);
+        _mm512_storeu_ps(a + i, v_a);
+        i += 16;
+    }
+
+    // process tails
+    if (i < size) {
+        __mmask16 mask = (1 << (size - i)) - 1;
+        v_a = _mm512_maskz_loadu_ps(mask, a + i);
+        v_b = _mm512_maskz_loadu_ps(mask, b + i);
+        v_a = _mm512_fmadd_ps(v_a, v_scale, v_b);
+        v_max = _mm512_mask_max_ps(v_max, mask, v_a, v_max);
+        _mm512_mask_storeu_ps(a + i, mask, v_a);
+    }
+
+    max = _mm512_reduce_max_ps(v_max);
+#elif defined(HAVE_AVX2)
+    auto v_max = _mm256_set1_ps(std::numeric_limits<float>::lowest());
+    auto v_scale = _mm256_set1_ps(scale);
+    auto v_a = v_max;
+    auto v_b = v_max;
+    size_t i = 0;
+    // process vector body
+    while (i + 7 < size) {
+        v_a = _mm256_loadu_ps(a + i);
+        v_b = _mm256_loadu_ps(b + i);
+        v_a = _mm256_fmadd_ps(v_a, v_scale, v_b);
+        v_max = _mm256_max_ps(v_max, v_a);
+        _mm256_storeu_ps(a + i, v_a);
+        i += 8;
+    }
+
+    // process tails
+    if (i < size) {
+        auto mask = get_mask(size - i);
+        v_a = _mm256_maskload_ps(a + i, mask);
+        v_b = _mm256_maskload_ps(b + i, mask);
+        v_a = _mm256_fmadd_ps(v_a, v_scale, v_b);
+        v_a = _mm256_blendv_ps(v_max, v_a, _mm256_castsi256_ps(mask));
+        v_max = _mm256_max_ps(v_max, v_a);
+        _mm256_maskstore_ps(a + i, mask, v_a);
+    }
+    hmax(v_max);
+    max = _mm256_cvtss_f32(v_max);
+#else
+    for (size_t i = 0; i < size; i++) {
+        a[i] *= scale;
+        a[i] += b[i];
+        max = a[i] > max ? a[i] : max;
+    }
+#endif
+}
+
+template <int dispatch_id>
+inline void scale_add2_reduce_max(float* a,
+                                  const float scale,
+                                  const float* alibi,
+                                  const float* attn_mask,
+                                  const uint8_t* causal_mask,
+                                  bool select_nfltmax_at_0,  // true:  0 in mask set -FLT_MAX
+                                  const size_t size,
+                                  float& max) {
+#if defined(HAVE_AVX512F)
+    auto v_max = _mm512_set1_ps(std::numeric_limits<float>::lowest());
+    auto v_scale = _mm512_set1_ps(scale);
+    auto v_a = v_max;
+    size_t i = 0;
+    auto v_zeroi32 = _mm512_setzero_epi32();
+    auto v_nfltmax = _mm512_set1_ps(-FLT_MAX);
+    auto kmask_xor = _cvtu32_mask16(select_nfltmax_at_0 ? 0xFFFF : 0);
+    // process vector body
+    while (i + 16 <= size) {
+        v_a = _mm512_loadu_ps(a + i);
+        v_a = _mm512_mul_ps(v_a, v_scale);
+
+        if (dispatch_id & 0x100) {
```

Comment:
Not sure that `dispatch_id` is the best solution. Maybe having 3 separate bool template parameters will improve the readability?

Reply:
Done.